x-pack/plugins/security/server/authentication/api_keys.ts

```diff
@@ -164,4 +164,19 @@ export class APIKeys {
 
     return result;
   }
+
+  async areEnabled(): Promise<boolean> {
+    if (!this.license.isEnabled()) {
+      return false;
+    }
+
+    // `transport.request` is potentially unsafe when combined with untrusted user input.
+    // Do not augment with such input.
+    const result = await this.clusterClient.callAsInternalUser('transport.request', {
+      method: 'GET',
+      path: '/_xpack/usage',
+    });
+
+    return result.security?.api_key_service?.enabled === true;
```

Comment:
__note:__ the only thing that worries me a bit here is that we don't have a proper ES documentation on the format of this response and it's not clear what BWC guarantees this API has. If format changes this line will "silently" start always returning `false`. 

Ideally I'd have an API integration test (e.g. via plugin-fixture that depends on `security` plugin, luckily we can have NP plugin fixtures now), but it seems we don't have any integration tests for the API keys functionality so it's not a blocker for this PR, up to you.

Reply:
That's a great point, let me add API integrations tests specifically for this.